chronos/asyncfutures2.nim

```diff
@@ -571,6 +571,48 @@ proc internalRead*[T](fut: Future[T]): T {.inline.} =
   when T isnot void:
     return fut.value
 
+macro checkFutureExceptions*(f, typ: typed): untyped =
```

Comment:
Whatever you are doing please leave all the macro related code in `asyncmacro2.nim`